src/view.c

```diff
@@ -168,49 +167,20 @@ view_move_resize(struct view *view, struct wlr_box geo)
 	}
 }
 
-#define MIN_VIEW_WIDTH (100)
-#define MIN_VIEW_HEIGHT (60)
-
-#if HAVE_XWAYLAND
-static int
-round_to_increment(int val, int base, int inc)
-{
-	if (base < 0 || inc <= 0)
-		return val;
-	return base + (val - base + inc / 2) / inc * inc;
-}
-#endif
-
 void
 view_adjust_size(struct view *view, int *w, int *h)
 {
 	assert(view);
-	int min_width = MIN_VIEW_WIDTH;
-	int min_height = MIN_VIEW_HEIGHT;
-#if HAVE_XWAYLAND
-	if (view->type == LAB_XWAYLAND_VIEW) {
-		xcb_size_hints_t *hints =
-			xwayland_surface_from_view(view)->size_hints;
 
-		/*
-		 * Honor size increments from WM_SIZE_HINTS. Typically, X11
-		 * terminal emulators will use WM_SIZE_HINTS to make sure that
-		 * the terminal is resized to a width/height evenly divisible by
-		 * the cell (character) size.
-		 */
-		if (hints) {
-			*w = round_to_increment(*w, hints->base_width,
-				hints->width_inc);
-			*h = round_to_increment(*h, hints->base_height,
-				hints->height_inc);
-
-			min_width = MAX(1, hints->min_width);
-			min_height = MAX(1, hints->min_height);
-		}
+#if HAVE_XWAYLAND
+	if (xwayland_apply_size_hints(view, w, h)) {
+		/* We don't want to cap the size to keep the aspect ratio */
```

Comment:
Do we really want this though? At least `GDK_BACKEND=x11 mate-terminal` can turn out smaller than the SSD which is kind of weird IMO.

Reply:
XWayland windows can set their own min size, and some are not decorated. For example Audacious can have windows "rolled up" to less than 60px high.

Reply:
That could work, no objections here.

Reply:
Agree with 4*BUTTON_WIDTH approach